### ISSUE TITLE
[ISSUE] Destroying databricks_workspace_conf fails

### ISSUE THREAD

**Issue by snapseb1** [databricks/terraform-provider-databricks#1229]
Hi,

Destroy on databricks_workspace_conf fails with maxTokenLifetimeDays set to 365 days.

### Configuration
```
resource "databricks_workspace_conf" "dbwconf" {
  count = lower(var.dbw_sku) == "premium" ? 1 : 0
  custom_config = {
    "enableIpAccessLists" : var.dbw_enableIpAccessLists
    "enableTokensConfig" : var.dbw_enableTokensConfig
    "maxTokenLifetimeDays" : var.dbw_maxTokenLifetimeDays
  }
}

### Expected Behavior
On destroy, the databricks workspace configuration should be destroyed without any error.

### Actual Behavior
The destroy fails with error: Some values are not allowed

  ```
# module.main.databricks_workspace_conf.dbwconf[0] will be destroyed
  - resource "databricks_workspace_conf" "dbwconf" {
      - custom_config = {
          - "enableIpAccessLists"  = "true"
          - "enableTokensConfig"   = "true"
          - "maxTokenLifetimeDays" = "365"
        } -> null
      - id            = "_" -> null
    }

│ Error: cannot delete workspace conf: Some values are not allowed: {"maxTokenLifetimeDays":""}

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

resource "databricks_workspace_conf" "dbwconf" {
  count = lower(var.dbw_sku) == "premium" ? 1 : 0
  custom_config = {
    "enableIpAccessLists" : "true"
    "enableTokensConfig" : "true"
    "maxTokenLifetimeDays" : "365"
  }
}

2. `terraform destory`

### Terraform and provider versions
Terraform v0.15.5
on darwin_amd64
+ provider registry.terraform.io/databrickslabs/databricks v0.5.4
+ provider registry.terraform.io/hashicorp/azuread v0.11.0
+ provider registry.terraform.io/hashicorp/azurerm v2.99.0
+ provider registry.terraform.io/infobloxopen/infoblox v2.1.0

**Comment by snapseb1** [comment]
Terraform apply debug log

```
2022-04-02T21:34:03.794Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.4/linux_amd64/terraform-provider-databricks_v0.5.4 args=[.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.4/linux_amd64/terraform-provider-databricks_v0.5.4]
path=.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.4/linux_amd64/terraform-provider-databricks_v0.5.4 pid=146
2022-04-02T21:34:03.795Z [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.4/linux_amd64/terraform-provider-databricks_v0.5.4
2022-04-02T21:34:03.807Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Databricks Terraform Provider (experimental)
2022-04-02T21:34:03.807Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 
2022-04-02T21:34:03.807Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Version 0.5.4
2022-04-02T21:34:03.807Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 
2022-04-02T21:34:03.807Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs
2022-04-02T21:34:03.807Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 
2022-04-02T21:34:03.807Z [INFO]  provider.terraform-provider-databricks_v0.5.4: configuring server automatic mTLS: timestamp=2022-04-02T21:34:03.807Z
address=/tmp/plugin290293215 network=unix timestamp=2022-04-02T21:34:03.850Z
path=.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.4/linux_amd64/terraform-provider-databricks_v0.5.4 pid=146
2022-04-02T21:34:04.733Z [DEBUG] ProviderTransformer: "module.main.databricks_workspace_conf.dbwconf[0] (destroy)" (*terraform.NodeDestroyResourceInstance) needs module.main.provider["registry.terraform.io/databrickslabs/databricks"]
2022-04-02T21:34:04.734Z [DEBUG] ProviderTransformer: "module.main.databricks_workspace_conf.dbwconf (expand)" (*terraform.nodeExpandApplyableResource) needs module.main.provider["registry.terraform.io/databrickslabs/databricks"]
2022-04-02T21:34:04.734Z [DEBUG] ProviderTransformer: "module.main.databricks_ip_access_list.dbwallowedlist (expand)" (*terraform.nodeExpandApplyableResource) needs module.main.provider["registry.terraform.io/databrickslabs/databricks"]
2022-04-02T21:34:04.734Z [DEBUG] ProviderTransformer: "module.main.databricks_ip_access_list.dbwallowedlist[0] (destroy)" (*terraform.NodeDestroyResourceInstance) needs module.main.provider["registry.terraform.io/databrickslabs/databricks"]
2022-04-02T21:34:04.734Z [DEBUG] ProviderTransformer: "module.main.databricks_cluster.main (expand)" (*terraform.nodeExpandApplyableResource) needs module.main.provider["registry.terraform.io/databrickslabs/databricks"]
2022-04-02T21:34:04.734Z [DEBUG] ProviderTransformer: "module.main.azurerm_databricks_workspace.main (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/hashicorp/azurerm"]
2022-04-02T21:34:04.734Z [DEBUG] ProviderTransformer: "module.main.azurerm_databricks_workspace.main (destroy)" (*terraform.NodeDestroyResourceInstance) needs provider["registry.terraform.io/hashicorp/azurerm"]
2022-04-02T21:34:04.734Z [DEBUG] ProviderTransformer: "module.main.databricks_cluster.main[\"117138-cluster01-117138\"] (destroy)" (*terraform.NodeDestroyResourceInstance) needs module.main.provider["registry.terraform.io/databrickslabs/databricks"]
2022-04-02T21:34:04.735Z [DEBUG] pruning unused provider["registry.terraform.io/databrickslabs/databricks"]
2022-04-02T21:34:04.739Z [DEBUG] ReferenceTransformer: "module.main.databricks_cluster.main (expand)" references: [module.main.local.clusters (expand)]
2022-04-02T21:34:04.739Z [DEBUG] ReferenceTransformer: "module.main.provider[\"registry.terraform.io/databrickslabs/databricks\"]" references: [module.main.azurerm_databricks_workspace.main (expand)]
2022-04-02T21:34:04.740Z [DEBUG] ReferenceTransformer: "module.main.output.databricks_workspace_name (expand)" references: [module.main.azurerm_databricks_workspace.main (expand)]
2022-04-02T21:34:04.740Z [DEBUG] ReferenceTransformer: "module.main.databricks_ip_access_list.dbwallowedlist (expand)" references: [module.main.var.dbw_sku (expand)]
2022-04-02T21:34:04.740Z [DEBUG] ReferenceTransformer: "module.main.output.databricks_workspace_tags (expand)" references: [module.main.azurerm_databricks_workspace.main (expand)]
2022-04-02T21:34:04.740Z [DEBUG] ReferenceTransformer: "module.main.output.databricks_cluster_objects (expand)" references: [module.main.databricks_cluster.main (expand)]
2022-04-02T21:34:04.740Z [DEBUG] ReferenceTransformer: "module.main.azurerm_databricks_workspace.main (expand)" references: []
2022-04-02T21:34:04.741Z [DEBUG] ReferenceTransformer: "module.main.databricks_workspace_conf.dbwconf (expand)" references: [module.main.var.dbw_sku (expand)]
2022-04-02T21:34:04.741Z [DEBUG] ReferenceTransformer: "module.main.output.databricks_workspace_id (expand)" references: [module.main.azurerm_databricks_workspace.main (expand)]
2022-04-02T21:34:04.741Z [DEBUG] ReferenceTransformer: "module.main.output.databricks_workspace_object (expand)" references: [module.main.azurerm_databricks_workspace.main (expand)]
2022-04-02T21:34:04.750Z [DEBUG] pruneUnusedNodes: module.main.output.databricks_workspace_name (expand) is no longer needed, removing
2022-04-02T21:34:04.750Z [DEBUG] pruneUnusedNodes: module.main.databricks_ip_access_list.dbwallowedlist (expand) is no longer needed, removing
2022-04-02T21:34:04.750Z [DEBUG] pruneUnusedNodes: module.main.output.databricks_workspace_tags (expand) is no longer needed, removing
2022-04-02T21:34:04.750Z [DEBUG] pruneUnusedNodes: module.main.output.databricks_cluster_objects (expand) is no longer needed, removing
2022-04-02T21:34:04.751Z [DEBUG] pruneUnusedNodes: module.main.databricks_workspace_conf.dbwconf (expand) is no longer needed, removing
2022-04-02T21:34:04.751Z [DEBUG] pruneUnusedNodes: module.main.output.databricks_workspace_id (expand) is no longer needed, removing
2022-04-02T21:34:04.751Z [DEBUG] pruneUnusedNodes: module.main.databricks_cluster.main (expand) is no longer needed, removing
2022-04-02T21:34:04.752Z [DEBUG] pruneUnusedNodes: module.main.output.databricks_workspace_object (expand) is no longer needed, removing
2022-04-02T21:34:04.752Z [DEBUG] pruneUnusedNodes: module.main.azurerm_databricks_workspace.main (expand) is no longer needed, removing
2022-04-02T21:34:05.096Z [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.4/linux_amd64/terraform-provider-databricks_v0.5.4 args=[.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.4/linux_amd64/terraform-provider-databricks_v0.5.4]
path=.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.4/linux_amd64/terraform-provider-databricks_v0.5.4 pid=252
2022-04-02T21:34:05.096Z [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.4/linux_amd64/terraform-provider-databricks_v0.5.4
2022-04-02T21:34:05.111Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Databricks Terraform Provider (experimental)
2022-04-02T21:34:05.111Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 
2022-04-02T21:34:05.111Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Version 0.5.4
2022-04-02T21:34:05.111Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 
2022-04-02T21:34:05.111Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs
2022-04-02T21:34:05.111Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 
2022-04-02T21:34:05.112Z [INFO]  provider.terraform-provider-databricks_v0.5.4: configuring server automatic mTLS: timestamp=2022-04-02T21:34:05.112Z
address=/tmp/plugin169522317 network=unix timestamp=2022-04-02T21:34:05.173Z
2022-04-02T21:34:05.257Z [WARN]  ValidateProviderConfig from "module.main.provider[\"registry.terraform.io/databrickslabs/databricks\"]" changed the config value, but that value is unused
2022-04-02T21:34:05.258Z [INFO]  provider.terraform-provider-databricks_v0.5.4: Explicit and implicit attributes: azure_client_id, azure_client_secret, azure_tenant_id, azure_workspace_resource_id: timestamp=2022-04-02T21:34:05.258Z
[0m[1mmodule.main.databricks_ip_access_list.dbwallowedlist[0]: Destroying... [id=f78e64ed-09cd-45de-b0a5-adc6c3726b6d][0m[0m
2022-04-02T21:34:05.260Z [INFO]  Starting apply for module.main.databricks_ip_access_list.dbwallowedlist[0]
2022-04-02T21:34:05.260Z [DEBUG] module.main.databricks_ip_access_list.dbwallowedlist[0]: applying the planned Delete change
2022-04-02T21:34:05.262Z [INFO]  provider.terraform-provider-databricks_v0.5.4: Generating AAD token for Azure Service Principal: timestamp=2022-04-02T21:34:05.262Z
2022-04-02T21:34:05.262Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Getting Workspace ID via management token.: timestamp=2022-04-02T21:34:05.262Z
[0m[1mmodule.main.databricks_cluster.main["117138-cluster01-117138"]: Destroying... [id=0402-212654-3vb1cxxf][0m[0m
2022-04-02T21:34:05.262Z [INFO]  Starting apply for module.main.databricks_cluster.main["117138-cluster01-117138"]
2022-04-02T21:34:05.262Z [DEBUG] module.main.databricks_cluster.main["117138-cluster01-117138"]: applying the planned Delete change
2022-04-02T21:34:05.499Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: GET /subscriptions/3c51cc28-d5df-44bb-b15c-cfdd64fff599/resourceGroups/RGTERRATEST-117138/providers/Microsoft.Databricks/workspaces/dbw-117138: timestamp=2022-04-02T21:34:05.499Z
2022-04-02T21:34:06.272Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 200 OK  {
    "managedResourceGroupId": "/subscriptions/3c51cc28-d5df-44bb-b15c-cfdd64fff599/resourceGroups/databricks-rg-RGTERRATEST-117... (3 more bytes)",
2022-04-02T21:34:06.273Z [INFO]  provider.terraform-provider-databricks_v0.5.4: Configured azure-client-secret auth: host=https://adb-1393290099077045.5.azuredatabricks.net/, azure_workspace_resource_id=/subscriptions/3c51cc28-d5df-44bb-b15c-cfdd64fff599/resourceGroups/RGTERRATEST-117138/providers/Microsoft.Databricks/workspaces/dbw-117138, azure_client_secret=***REDACTED***, azure_client_id=f16778ae-0039-48fc-8689-574da96273ab, azure_tenant_id=331d1275-437c-4934-8eae-f5c7a0170487: timestamp=2022-04-02T21:34:06.272Z
2022-04-02T21:34:06.273Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2022-04-02T21:34:06.272Z
2022-04-02T21:34:06.340Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2022-04-02T21:34:06.340Z
2022-04-02T21:34:06.384Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: DELETE /api/2.0/ip-access-lists/f78e64ed-09cd-45de-b0a5-adc6c3726b6d {}: timestamp=2022-04-02T21:34:06.384Z
2022-04-02T21:34:06.384Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: POST /api/2.0/clusters/delete {
2022-04-02T21:34:06.514Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 200 OK  {} <- POST /api/2.0/clusters/delete: timestamp=2022-04-02T21:34:06.514Z
2022-04-02T21:34:06.514Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Waiting for state to become: [success]: timestamp=2022-04-02T21:34:06.514Z
2022-04-02T21:34:06.514Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2022-04-02T21:34:06.514Z
2022-04-02T21:34:06.514Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: GET /api/2.0/clusters/get: timestamp=2022-04-02T21:34:06.514Z
2022-04-02T21:34:06.540Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 200 OK  {
2022-04-02T21:34:06.540Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Cluster 0402-212654-3vb1cxxf is TERMINATING: Termination requested by f16778ae-0039-48fc-8689-574da96273ab: timestamp=2022-04-02T21:34:06.540Z
2022-04-02T21:34:06.584Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 200 OK  {} <- DELETE /api/2.0/ip-access-lists/f78e64ed-09cd-45de-b0a5-adc6c3726b6d: timestamp=2022-04-02T21:34:06.584Z
[0m[1mmodule.main.databricks_ip_access_list.dbwallowedlist[0]: Destruction complete after 2s[0m
[0m[1mmodule.main.databricks_workspace_conf.dbwconf[0]: Destroying... [id=_][0m[0m
2022-04-02T21:34:06.586Z [INFO]  Starting apply for module.main.databricks_workspace_conf.dbwconf[0]
2022-04-02T21:34:06.586Z [DEBUG] module.main.databricks_workspace_conf.dbwconf[0]: applying the planned Delete change
2022-04-02T21:34:06.587Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2022-04-02T21:34:06.587Z
2022-04-02T21:34:06.588Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: PATCH /api/2.0/workspace-conf {
2022-04-02T21:34:06.621Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 400 Bad Request {
2022-04-02T21:34:06.621Z [WARN]  provider.terraform-provider-databricks_v0.5.4: /api/2.0/workspace-conf:400 - Some values are not allowed: {"maxTokenLifetimeDays":""}: timestamp=2022-04-02T21:34:06.621Z
2022-04-02T21:34:06.621Z [WARN]  provider.terraform-provider-databricks_v0.5.4: /api/2.0/workspace-conf:400 - Some values are not allowed: {"maxTokenLifetimeDays":""}: timestamp=2022-04-02T21:34:06.621Z
2022-04-02T21:34:07.041Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2022-04-02T21:34:07.041Z
2022-04-02T21:34:07.041Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: GET /api/2.0/clusters/get: timestamp=2022-04-02T21:34:07.041Z
2022-04-02T21:34:07.093Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 200 OK  {
2022-04-02T21:34:07.093Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Cluster 0402-212654-3vb1cxxf is TERMINATING: Termination requested by f16778ae-0039-48fc-8689-574da96273ab: timestamp=2022-04-02T21:34:07.093Z
2022-04-02T21:34:08.094Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2022-04-02T21:34:08.094Z
2022-04-02T21:34:08.094Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: GET /api/2.0/clusters/get: timestamp=2022-04-02T21:34:08.094Z
2022-04-02T21:34:08.123Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 200 OK  {
2022-04-02T21:34:08.123Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Cluster 0402-212654-3vb1cxxf is TERMINATING: Termination requested by f16778ae-0039-48fc-8689-574da96273ab: timestamp=2022-04-02T21:34:08.123Z
2022-04-02T21:34:10.124Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2022-04-02T21:34:10.123Z
2022-04-02T21:34:10.124Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: GET /api/2.0/clusters/get: timestamp=2022-04-02T21:34:10.123Z
2022-04-02T21:34:10.168Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 200 OK  {
2022-04-02T21:34:10.168Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Cluster 0402-212654-3vb1cxxf is TERMINATING: Termination requested by f16778ae-0039-48fc-8689-574da96273ab: timestamp=2022-04-02T21:34:10.168Z
2022-04-02T21:34:14.170Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2022-04-02T21:34:14.170Z
2022-04-02T21:34:14.170Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: GET /api/2.0/clusters/get: timestamp=2022-04-02T21:34:14.170Z
2022-04-02T21:34:14.198Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 200 OK  {
2022-04-02T21:34:14.198Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Cluster 0402-212654-3vb1cxxf is TERMINATING: Termination requested by f16778ae-0039-48fc-8689-574da96273ab: timestamp=2022-04-02T21:34:14.198Z
[0m[1mmodule.main.databricks_cluster.main["117138-cluster01-117138"]: Still destroying... [id=0402-212654-3vb1cxxf, 10s elapsed][0m[0m
2022-04-02T21:34:22.204Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2022-04-02T21:34:22.204Z
2022-04-02T21:34:22.204Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: GET /api/2.0/clusters/get: timestamp=2022-04-02T21:34:22.204Z
2022-04-02T21:34:22.232Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 200 OK  {
2022-04-02T21:34:22.232Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Cluster 0402-212654-3vb1cxxf is TERMINATED: Termination requested by f16778ae-0039-48fc-8689-574da96273ab: timestamp=2022-04-02T21:34:22.232Z
2022-04-02T21:34:22.271Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: Setting 'X-Databricks-Azure-SP-Management-Token' header: timestamp=2022-04-02T21:34:22.271Z
2022-04-02T21:34:22.271Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: POST /api/2.0/clusters/permanent-delete {
2022-04-02T21:34:22.351Z [DEBUG] provider.terraform-provider-databricks_v0.5.4: 200 OK  {} <- POST /api/2.0/clusters/permanent-delete: timestamp=2022-04-02T21:34:22.351Z
[0m[1mmodule.main.databricks_cluster.main["117138-cluster01-117138"]: Destruction complete after 17s[0m
path=.terraform/providers/registry.terraform.io/databrickslabs/databricks/0.5.4/linux_amd64/terraform-provider-databricks_v0.5.4 pid=252

**Comment by snapseb1** [comment]
Please provide details about why this won't be fixed

**Comment by venkatasai-vanaparthi** [comment]
Hey @alexott , as per our discussion I will ask customer to re-open this issue

**Comment by venkatasai-vanaparthi** [comment]
Hi @snapseb1 , can you re-open this issue?

**Comment by snapseb1** [comment]
Re-opening the issue as its not fixed even in Version 1.0.0

**Comment by snapseb1** [comment]
@nfx @alexott  @venkatasai-vanaparthi  
It is not allowing me to re-open the issue. Could you re-open the issue?
